mailbox/maildir/src/main/resources/META-INF/spring/mailbox-maildir.xml

```diff
@@ -1,67 +1,68 @@
-        <constructor-arg index="0" ref="maildirStore"/>
-    </bean>
-    <bean id="maildirStore" class="org.apache.james.mailbox.maildir.MaildirStore">
-        <constructor-arg index="0" value="${maildir.folder}"/>
-        <property name="messageNameStrictParse" value="${maildir.messageNameParser.strictMode}"/>
-    </bean>
-    <alias name="jvm-locker" alias="maildir-locker"/>
-
-    <bean id="maildir-mailbox-id-deserializer" class="org.apache.james.mailbox.maildir.MaildirMailboxIdDeserializer"/>
-
-</beans>
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing,
+  software distributed under the License is distributed on an
+  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+  KIND, either express or implied.  See the License for the
+  specific language governing permissions and limitations
+  under the License.    
+-->
+
+<beans xmlns="http://www.springframework.org/schema/beans"
+       xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+       xsi:schemaLocation="
+          http://www.springframework.org/schema/beans http://www.springframework.org/schema/beans/spring-beans.xsd">
+
+    <!-- 
+      Mailbox Maildir
+      WARNING: Maildir does only work on UNIX like operation systems
+     -->
+    <bean id="maildir-mailboxIdFactory" class="org.apache.james.mailbox.maildir.MaildirId.Factory" />
+
+    <bean class="org.springframework.beans.factory.config.PropertyPlaceholderConfigurer">
+        <property name="ignoreUnresolvablePlaceholders" value="true"/>
+        <property name ="location" value="classpath:mailbox-maildir.properties"/>
+    </bean>
+
+    <bean id="maildir-mailboxmanager" class="org.apache.james.mailbox.store.StoreMailboxManager">
+        <constructor-arg index="0" ref="maildir-sessionMapperFactory"/>
+        <constructor-arg index="1" ref="sessionProvider"/>
+        <constructor-arg index="2" ref="memory-locker"/>
+        <constructor-arg index="3" ref="messageParser"/>
+        <constructor-arg index="4" ref="messageIdFactory" />
+        <constructor-arg index="5" ref="storeMailboxAnnotationManager" />
+        <constructor-arg index="6" ref="event-bus" />
+        <constructor-arg index="7" ref="storeRightManager" />
+        <constructor-arg index="8" ref="quotaComponents" />
+        <constructor-arg index="9" ref="indexer" />
+        <constructor-arg index="10" ref="no-mailbox-configuration" />
+        <constructor-arg index="11"><set/></constructor-arg>
```

Comment:
Should'nt we need a ref like for the other arguments?

Reply:
No if it works it's cool. (its different because we inject a set @Arsnael ) Thanks @trantienduchn for searching Spring graveyard and doing findings like this one. It might solve some other similar problems as well.